weed/filer/filer_rename.go

```diff
@@ -2,11 +2,37 @@ package filer
 
 import (
 	"fmt"
-	"github.com/seaweedfs/seaweedfs/weed/util"
+	"reflect"
 	"strings"
+
+	"github.com/seaweedfs/seaweedfs/weed/util"
 )
 
-func (f *Filer) CanRename(source, target util.FullPath) error {
+func (f *Filer) CanRename(source, target util.FullPath, oldName string) error {
+	sourcePath := strings.Split(string(source), "/")
+	if sourcePath[len(sourcePath)-1] == "" && len(sourcePath) > 0 {
+		sourcePath = sourcePath[:len(sourcePath)-1]
+	}
+	targetPath := strings.Split(string(target), "/")
+	sourceLen := len(sourcePath)
+	targetLen := len(targetPath)
+
+	if targetLen > sourceLen { // target is subdirectory
```

Comment:
just check strings.HasPrefix(target, source)?

Reply:
Thanks for your suggestion, I have modified it